template/.github/workflows/ci.yml

```diff
@@ -170,3 +170,60 @@ jobs:
 
     - name: Run the functional test suite
       run: poetry run invoke test --suite=functional
+
+build-and-test-docker-image:
+    name: Build and Test Docker Image
+    needs: [static-analysis-checks, type-checking, run-system-tests, run-functional-tests]
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_USER: dev
+          POSTGRES_PASSWORD: dev_password
+          POSTGRES_DB: dev
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+            # Allows us to use the same env config used locally by mapping to port to
+            # the test runner container.
+            - 5432:5432
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install dependencies
+      run: |
+        pip install invoke
+        cp example.env .env
+
+    - name: Modify DATABASE_URL for GitHub Actions
+      run: |
+        echo 'DATABASE_URL=postgres://dev:dev_password@postgres:5432/dev'  >> .env
+        echo 'DJANGO_SETTINGS_MODULE=avocado.main.settings' >> .env
+        cat .env
+
+    - name: Build Docker Image
+      run: invoke build-image --tag abc
+
+    - name: Run Docker image
+      run: |
+        docker run --name test-container --env-file .env avocado:abc
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** Where did the image name `test-container` come from?

I'm a bit confused where the image name `test-container` comes from. The `build-image` task _should_ build an image with the name `{ PACKAGE }`. I could understand that you might be using that name in your own test project, but `PACKAGE` _has_ to be a valid python package name - which would use `_` not `-`. ‏

Reply:
`test-container` was intended to be name of the container. It was part of my trial approach to split running the docker image itself and execute the django check command on when debugging.

Reply:
ok, that makes sense. So it won't be necessary when you write an invoke task to either run the container or return the `container:tag`.

Reply:
I added a task to run the docker image.